### PR TITLE
Lightbox

### PR DESCRIPTION
**Pull request by Sam1301**
Based on #660 
* uses react-native-photo-view for zoom.

brief video
https://drive.google.com/file/d/0B4jK5QX65b0ZbWVqY1VJMFZuVnc/view?usp=sharing

I would add a shared element transition or a vertical modal transition for lightbox once react navigation pr is merged.

### CODE REVIEW

File: src/html/tags/HtmlTagImg.js [hunk 1]
```diff
@@ -10,10 +9,17 @@ const styles = StyleSheet.create({
...
-  <Image
-    source={getResource(src, auth)}
-    resizeMode="contain"
-    style={[styles.img, style]}
-  />
-);
+export default ({ src, style, auth, message, pushRoute }) => {
+  const resource = getResource(src, auth);
+  return (
+    <TouchableWithoutFeedback
```
Comment (borisyankov) [inline src/html/tags/HtmlTagImg.js]: Do we want it to be without feedback?

File: src/html/tags/HtmlTagImg.js [hunk 2]
```diff
@@ -10,10 +9,17 @@ const styles = StyleSheet.create({
...
+      onPress={
```
Comment (borisyankov) [inline src/html/tags/HtmlTagImg.js]: Extract that to a this.handlePress and convert the component to a class.

File: src/lightbox/LightboxContainer.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  },
+  container: {
+    flex: 1,
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+});
+
+export default connectActionSheet(
+  connect(null, boundActions)(
```
Comment (borisyankov) [inline src/lightbox/LightboxContainer.js]: Split the connects from the class like the other components (first the class, then the export with the connect)

File: src/lightbox/LightboxContainer.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    class LightboxContainer extends React.Component {
+      state = {
+        movement: 'out',
+      };
+
+      handleImagePress = () => {
+        this.setState({
+          movement: this.state.movement === 'out' ? 'in' : 'out',
```
Comment (borisyankov) [inline src/lightbox/LightboxContainer.js]: When a state depends on its previous value use the setState(function) variation (see the help, it is not a well known feature)

File: src/styles/index.js [hunk 5]
```diff
@@ -5,7 +5,9 @@ export {
   STATUSBAR_HEIGHT,
   NAVBAR_HEIGHT,
   REACTION_HEIGHT,
-  REACTION_SPINNER_OFFSET
+  REACTION_SPINNER_OFFSET,
+  LIGHTBOX_FOOTER_OFFSET,
+  LIGHTBOX_OVERLAY_COLOR,
```
Comment (Sam1301) [inline src/styles/index.js]: due to perf issues, avoided alpha compositing on android.

File: package.json [hunk 6]
```diff
@@ -41,8 +41,11 @@
     "react-native-drawer": "^2.3.0",
     "react-native-fabric": "^0.4.1",
     "react-native-notifications": "^1.1.10",
+    "react-native-fetch-blob": "^0.10.5",
+    "react-native-photo-view": "^1.2.0",
     "react-native-safari-view": "^2.0.0",
     "react-native-scrollable-tab-view": "^0.7.0",
+    "react-native-simple-toast": "0.0.6",
```
Comment (Sam1301) [inline package.json]: @vishwesh3 used your commit f598ae671138e98be15f39d80555338ff80191b4  from #651, thanks :)

File: src/html/tags/HtmlTagImg.js (hunk 1 above)
Comment (Sam1301) [inline src/html/tags/HtmlTagImg.js]: used Touchable for images and without feedback for emojis

File: src/lightbox/LightboxContainer.js (hunk 4 above)
Comment (Sam1301) [inline src/lightbox/LightboxContainer.js]: nice, really like this simplification ;)

**Comment by borisyankov** [comment]
Hey, I merged your previous PR and this one got a lot of conflicts.
Please rebase and it is ready to merge too.

**Comment by borisyankov** [comment]
You are fast 👍

**Pull request merged**